backend/translation/views.py

```diff
@@ -1,199 +1,246 @@
-                required=True,
+                description="An integer identifying the translation instance",
             ),
-            openapi.Parameter(
-                "target_lang",
-                openapi.IN_QUERY,
-                description=("A string to pass the target language of the translation"),
+            "target_language": openapi.Schema(
                 type=openapi.TYPE_STRING,
-                required=True,
+                description="A string to pass the target language of the translation",
             ),
-            openapi.Parameter(
-                "get_latest",
-                openapi.IN_QUERY,
-                description=(
-                    "A string to pass whether to get the latest translation or not"
-                ),
+            "payload": openapi.Schema(
                 type=openapi.TYPE_STRING,
-                required=True,
+                description="A string to pass the translated subtitles and metadata",
+            ),
+            "transcript_id": openapi.Schema(
+                type=openapi.TYPE_STRING,
+                description="A string to pass the transcript uuid",
             ),
-        ],
-        responses={
-            200: "Generates the translation for the given transcript_id and target_lang"
         },
-    )
-    @api_view(["GET"])
-    def get(self, request):
-        # Get the query params
-        transcript_id = request.query_params.get("transcript_id")
-        target_lang = request.query_params.get("target_lang")
-        get_latest = request.query_params.get("get_latest")
+        description="Post request body",
+    ),
+    responses={
+        200: "Translation has been created/updated successfully",
+        400: "Bad request",
+        404: "No translation found for the given transcript_id and target_language",
+    },
+)
+@api_view(["POST"])
+@permission_classes([IsAuthenticated])
+def save_translation(request):
+    
+    # Get the required data from the POST body
+    translation_id = request.data.get("translation_id", None)
+    target_language = request.data["target_language"]
+    payload = request.data["payload"]
+    user = request.user
+
+    # If translation_id is not present, save a new translation object 
+    if translation_id is None:
+        
+        # Check if transcript_id is present in the POST body
+        if "transcript_id" not in request.data:
+            return Response(
+                {"error": "Transcript_id is missing from the POST body, which needs to be passed if translation_id is not passed."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        
+        # Get the transcript_id from the POST body
+        transcript_id = request.data["transcript_id"]
 
         # Ensure that the UUID is valid
         if not validate_uuid4(transcript_id):
             return Response(
                 {"error": "Invalid transcript_id."}, status=status.HTTP_400_BAD_REQUEST
             )
```

Comment:
We can remove this as well, as discussed earlier.

Reply:
Removed.